subsys/sensing/sensor/phy_3d_sensor/phy_3d_sensor.c

```diff
@@ -18,65 +19,222 @@
 
 LOG_MODULE_REGISTER(phy_3d_sensor, CONFIG_SENSING_LOG_LEVEL);
 
-static int phy_3d_sensor_init(const struct device *dev,
-		const struct sensing_sensor_info *info,
-		const sensing_sensor_handle_t *reporter_handles,
-		int reporters_count)
+#define SENSING_ACCEL_Q31_SHIFT 6
+#define SENSING_GYRO_Q31_SHIFT 15
+
+static int64_t shifted_q31_to_scaled_int64(q31_t q, int8_t shift, int64_t scale)
 {
-	return 0;
+	int64_t scaled_value;
+	int64_t shifted_value;
+
+	shifted_value = (int64_t)q << shift;
```

Comment:
You need to be careful with this it caught me off guard a few times, if shift is negative this will not work as expected. you need to left shift on positive and right shift by the absolute value on negative.

Reply:
Thanks for your reminder. This function can be removed after using `sensor_read` and `decoder` instead of `fetch/get`